src/targets/gpu/include/migraphx/gpu/gemm.hpp

```diff
@@ -75,6 +75,7 @@ struct rocblas_gemm
 
     shape compute_shape(const std::vector<shape>& inputs) const
     {
+        check_shapes{inputs, *this}.not_broadcasted();
```

Comment:
This is too broad and will have a negative impact on performance.

Reply:
I missed that. I modified the solution to only check the last two dimensions. I also moved the check to `blas_shape`, which I believe is a better place for it than directly in `compute_shape`.